packages/react-reconciler/src/__tests__/ReactSuspenseList-test.internal.js

```diff
@@ -1027,4 +1027,66 @@ describe('ReactSuspenseList', () => {
       </Fragment>,
     );
   });
+
+  it('switches to rendering fallbacks if the tail takes long CPU time', async () => {
+    function Foo() {
+      return (
+        <SuspenseList revealOrder="forwards">
+          <Suspense fallback={<Text text="Loading A" />}>
+            <Text text="A" />
+          </Suspense>
+          <Suspense fallback={<Text text="Loading B" />}>
+            <Text text="B" />
+          </Suspense>
+          <Suspense fallback={<Text text="Loading C" />}>
+            <Text text="C" />
+          </Suspense>
+        </SuspenseList>
+      );
+    }
+
+    // This render is only CPU bound. Nothing suspends.
+    ReactNoop.render(<Foo />);
+
+    expect(Scheduler).toFlushAndYieldThrough(['A']);
+
+    Scheduler.advanceTime(300);
+    jest.advanceTimersByTime(300);
+
+    expect(Scheduler).toFlushAndYieldThrough(['B']);
+
+    Scheduler.advanceTime(300);
+    jest.advanceTimersByTime(300);
+
+    // We've still not been able to show anything on the screen even though
+    // we have two items ready.
+    expect(ReactNoop).toMatchRenderedOutput(null);
+
+    // Time has now elapsed for so long that we're just going to give up
+    // rendering the rest of the content. So that we can at least show
+    // something.
+    expect(Scheduler).toFlushAndYieldThrough([
+      'Loading C',
+      'C', // I'll flush through into the next render so that the first commits.
```

Comment:
This is a helpful inline comment. In general, I find over-flushing approaches like this confusing at first glance.

One thing that I think maybe reads a little clearer (though it's more verbose) is to yield in an effect:
https://github.com/facebook/react/blob/34ce57ae751e0952fd12ab532a3e5694445897ea/packages/react/src/__tests__/ReactDOMTracing-test.internal.js#L232-L236

Then flush through the effect, rather than the first part of the next render:
https://github.com/facebook/react/blob/34ce57ae751e0952fd12ab532a3e5694445897ea/packages/react/src/__tests__/ReactDOMTracing-test.internal.js#L265

Just sharing, obviously no big deal :D